Security/Web_Security.mediawiki

```diff
@@ -0,0 +1,614 @@
+
+Redirections should be done with the 301 redirects, unless they redirect to a different path, in which case they may be done with 302 redirections. Sites should avoid redirections from HTTP to HTTPS on a different host, as this prevents HSTS from being set.
+
+=== Examples ===
+
+<pre># Redirect from http to https with nginx
+server {
+  listen 80;
+  server_name site.mozilla.org;
+
+  return 301 https://$server_name$request_uri;
+}</pre>
+
+<pre># Redirect from http to https with Apache
+<VirtualHost *:80>
+  ServerName site.mozilla.org
+  Redirect permanent / https://site.mozilla.org/
+</VirtualHost></pre>
+
+
+== HTTP Public Key Pinning ==
+
+Maximum risk sites must enable the use of HTTP Public Key Pinning (HPKP). HPKP instructs a user agent to bind a site to specific root certificate authority, intermediate certificate authority, or end-entity public key. This prevents  certificate authorities from issuing unauthorized certificates for a given domain that would nevertheless be trusted by the browsers. These fradulent certificates would allow an active attacker to MitM and impersonate a website, intercepting credentials and other sensitive data.
+
+Due to the risk of knocking yourself off the internet, HPKP must be implemented with extreme care. This includes having backup key pins, testing on a non-production domain, testing with <tt>Public-Key-Pins-Report-Only</tt> and then finally doing initial testing with a very short-lived <tt>max-age</tt> directive. Because of the risk of creating a self-denial-of-service and the very low risk of a fraudulent certificate being issued, it is <em>not recommended</em> for the majority websites to implement HPKP.
+
+=== Directives ===
+
+* <tt>max-age:</tt> how long the user agent the keys will be pinned; the site must use a cert that meets these pins until this time expires
+* <tt>includeSubDomains:</tt> whether user agents should pin all subdomains to the same pins
+
+Unlike with HSTS, what to set <tt>max-age</tt> is highly individualized to a given site.  A longer value is more secure, but screwing up your key pins will result in your site being unavailable for a longer period of time. Recommended values fall between 15 and 120 days.
+
+=== Examples ===
+
+<pre># Pin to DigiCert, Let's Encrypt, and the local public-key, including subdomains, for 15 days
+Public-Key-Pins: max-age=1296000; includeSubDomains; pin-sha256="WoiWRyIOVNa9ihaBciRSC7XHjliYS9VwUGOIud4PB18=";
```

Comment:
l121: how do you get the pin-sha256? ( i mean, this has to be explained somehow)